src/Corporation/Actions.ts

```diff
@@ -345,10 +347,17 @@ export function PurchaseWarehouse(corp: ICorporation, division: IIndustry, city:
   corp.funds = corp.funds - CorporationConstants.WarehouseInitialCost;
 }
 
-export function UpgradeWarehouse(corp: ICorporation, division: IIndustry, warehouse: Warehouse): void {
-  const sizeUpgradeCost = CorporationConstants.WarehouseUpgradeBaseCost * Math.pow(1.07, warehouse.level + 1);
+export function UpgradeWarehouseCost(warehouse: Warehouse, amt: number): number {
+  return Array.from(Array(amt).keys()).reduce(
+    (acc, index) => acc + CorporationConstants.WarehouseUpgradeBaseCost * Math.pow(1.07, warehouse.level + 1 + index),
+    0,
+  );
+}
```

Comment:
```js
let cost = 0;
for(let n = 1; n <= amt; n++)
  cost += CorporationConstants.WarehouseUpgradeBaseCost * Math.pow(1.07, warehouse.level + n);
return cost;
```
Would IMO be more readable.

Doing it with a functional approach would require a few extra basic functions in order to make the code straightforward:
`interval(1. amt).map(n => CorporationConstants.WarehouseUpgradeBaseCost * Math.pow(1.07, warehouse.level + n).sum()`